src/main/java/com/igot/cb/cache/AccessSettingRuleCacheMgr.java

```diff
@@ -204,4 +206,107 @@ BitSet createBitSetForAttribute(Collection<Integer> attributeValues) {
         }
         return bitSet;
     }
+
+
+    public CachedAccessSettingRule getOrLoadAccessSettingRule(String courseId, String contextId) {
+        String cacheKey = courseId + "|" + contextId;
+
+        if (MapUtils.isEmpty(cachedAccessSettingRules)) {
+            log.info("Cache not initialized. Loading rules from Cassandra and Redis...");
+            try {
+                loadRulesFromCassandraAndCache();
+            } catch (Exception e) {
+                log.error("Error while initializing cache from Cassandra: {}", e.getMessage(), e);
+            }
+        }
+        CachedAccessSettingRule rule = cachedAccessSettingRules.get(cacheKey);
+        if (rule != null) {
+            log.debug("Cache hit for rule key: {}", cacheKey);
+            return rule;
+        }
+        log.info("Cache miss for rule key: {}, loading from Cassandra", cacheKey);
+        try {
+            Map<String, Object> filter = new HashMap<>();
+            filter.put(Constants.CONTEXT_ID, contextId);
+            filter.put(Constants.COURSE_ID, courseId);
+
+            List<Map<String, Object>> records = cassandraOperation.getRecordsByProperties(
+                    Constants.KEYSPACE_SUNBIRD_COURSE,
+                    Constants.ACCESS_SETTINGS_RULES_TABLE_V2,
+                    filter,
+                    null,
+                    null
+            );
+
+            if (CollectionUtils.isEmpty(records)) {
+                log.warn("No rule found in Cassandra for key: {}", cacheKey);
+                return null;
+            }
+
+            Map<String, Object> record = records.get(0);
+            CachedAccessSettingRule loadedRule = new CachedAccessSettingRule(
+                    (String) record.get(Constants.CONTEXT_ID),
+                    (String) record.get(Constants.CONTEXT_ID_TYPE),
+                    (String) record.get(Constants.CONTEXT_DATA),
+                    false
+            );
+
+            try {
+                Map<String, Object> contextData = loadedRule.getContextData();
+                if (MapUtils.isNotEmpty(contextData)) {
+                    processContextData(cacheKey, contextData);
+                }
+                cachedAccessSettingRules.put(cacheKey, loadedRule);
+                log.info("Loaded and cached rule for key: {}", cacheKey);
+            } catch (Exception e) {
+                log.error("Error processing rule {}: {}", cacheKey, e.getMessage(), e);
+            }
+
+            return loadedRule;
+
+        } catch (Exception e) {
+            log.error("Failed to load rule from Cassandra for key {}: {}", cacheKey, e.getMessage(), e);
+            return null;
+        }
+    }
+
+
+    private void loadRulesFromCassandraAndCache() {
```

Comment:
This is wrong. You should not attempt to load the access_settings_rule_v2 table all together at once.
The idea is to read this table on demand. Not to initialize these records always.
In this requirement too, you have list of courses which you need to check for access against a user. so call the getOrLoadAccessSettingRule in a loop for each courseid